eras/shelley/impl/testlib/Test/Cardano/Ledger/Shelley/CDDL.hs

```diff
@@ -83,13 +89,13 @@ operational_cert =
 next_major_protocol_version :: Integer
 next_major_protocol_version = 3
 
-major_protocol_version :: Rule
-major_protocol_version = "major_protocol_version" =:= (1 :: Integer) ... next_major_protocol_version
+major_protocol_version :: forall era. Era era => Rule
+major_protocol_version = "major_protocol_version" =:= (1 :: Integer) ... getVersion @Integer (eraProtVerHigh @era)
```

Comment:
Byron actually started at `0`. To make it clear we could actually use the `eraProtVerLow` of Byron era. Also for each era we allow a major protocol version that the next era will start in, otherwise we would not be able to upgrade current era to the next one.
```suggestion
major_protocol_version = "major_protocol_version" =:= (getVersion @Integer (eraProtVerLOw @ByronEra) ... succ (getVersion @Integer (eraProtVerHigh @era))
```